plugins/janus_audiobridge.c

```diff
@@ -875,16 +883,27 @@ static guint32 janus_audiobridge_rtp_forwarder_add_helper(janus_audiobridge_room
 	rf->seq_number = 0;
 	rf->timestamp = 0;
 	rf->always_on = always_on;
+
 	janus_mutex_lock(&room->rtp_mutex);
-	guint32 stream_id = janus_random_uint32();
-	while(g_hash_table_lookup(room->rtp_forwarders, GUINT_TO_POINTER(stream_id)) != NULL) {
-		stream_id = janus_random_uint32();
+
+	guint32 actual_stream_id;
+	if(stream_id > 0) {
+		actual_stream_id = stream_id;
```

Comment:
Should we actually check if the stream ID exists, and in case fail? As if it does, the code currently overwrites it with a random one, thus replacing whatever the user provided. Anyway, this is currently only allowed for the static RTP forwarder, and not for requests via the API, so we can safely ignore for the moment.

Reply:
It does the check, see the code below. If the ID exists, it will get a random ID. But as you mentioned, configuring a static RTP forwarder is always the first one. Thanks for merging!

Reply:
Yep, that was my point: using a random ID instead of what the user asked for is possibly a broken behaviour, and the request should fail instead. But since we don't currently allow users to specify a stream ID when adding a forwarder via API, it's a non-issue at the moment. I'll make sure this is addressed should it change in the future.